types/src/discovery_set.rs

```diff
@@ -124,10 +128,49 @@ impl IntoIterator for DiscoverySet {
     }
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct DiscoverySetChangeEventInternal {
+    pub discovery_set: DiscoverySet,
+    pub validator_set: ValidatorSet,
+}
+
+impl DiscoverySetChangeEventInternal {
+    pub fn try_from_bytes(bytes: &[u8]) -> Result<Self> {
+        lcs::from_bytes(bytes).map_err(Into::into)
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DiscoveryInfoFull {
```

Comment:
I would probably call the on chain data FullnodeDiscoveryInfo
and then this would just be DiscoveryInfo